### ISSUE TITLE
Add docs for using Cert Manager to manage Gateway Certs

### ISSUE THREAD

**Issue by arkodg** [envoyproxy/gateway#954]
*Description*:
>Describe the issue.

Add docs highlighting cert manager Gateway API integration https://cert-manager.io/docs/usage/gateway/

[optional *Relevant Links*:]
>Any extra documentation required to understand the issue.

**Comment by robd003** [comment]
Just want to make sure this isn't forgotten for the 0.4.0 release. Very interested in doing automatic certificate deployment for new hosts using cert-manager

**Comment by robd003** [comment]
One very specific question I had was if Gateway would automatically remove expired certificates if Envoy Gateway was the initial requestor of the certificate.

Example usage: every year an event is put on and a subdomain is set up for the event, however after the event is over the subdomain will never be used again. It would be great if certificates were automatically removed if there's no reference to them.

Having a bunch of dangling certificates that can't be renewed would be tedious to have to constantly clean up.

**Comment by tommie** [comment]
I was looking through cert-manager docs and code, and couldn't find anything about cleaning up garbage, except for [`CertificateRequest` history](https://github.com/cert-manager/cert-manager/blob/master/pkg/controller/certificates/revisionmanager/revisionmanager_controller.go). IMHO, garbage collection should be part of cert-manager, since references from other secrets is something Envoy couldn't know about in the general case.

More specifically, I'd have expected it to remove expired certificates, and not renew ones that aren't referenced in `Secret`s.

**Comment by robd003** [comment]
Any progress on this since last month?

**Comment by tommie** [comment]
Re. cleanup again

cert-amanger removes "unrequired" certificates it owns: https://github.com/cert-manager/cert-manager/blob/c5e6bf39d688d202592318eaf91988466a7ee37b/pkg/controller/certificate-shim/sync.go#L171

So I think this functionality exists, but is up to cert-manager.

Also: https://cert-manager.io/docs/usage/certificate/#cleaning-up-secrets-when-certificates-are-deleted

I.e. cert-manager can be configured to also remove Secrets it generates based on Certificates being removed. However, note that these are usually manually referenced in `Gateway` listener configs. Not sure if cert-manager should do that cleanup, or not, but I'm also not sure it matters. If there is no cert referencing the secret, it's probably because the Gateway was removed?

**Comment by tommie** [comment]
@robd003 I've documented what I did to get cert-manager running: https://github.com/envoyproxy/gateway/pull/1461/files

Untested:

* My script isn't using Helm to install cert-manager (yet,) so I might have that wrong.
* I haven't tried the Let's Encrypt production environment, but it should be fine.
* I haven't looked into garbage collection that much. I've documented all the Gateway API-specific aspects, I think.
